features/steps/reusable_steps.rb

```diff
@@ -6,24 +6,20 @@
   Maze.config.captured_invalid_requests.delete(type.to_sym)
 end
 
-Then(/^on (iOS|macOS|watchOS), (.+)/) do |platform, step_text|
-  step(step_text) if platform.downcase == Maze::Helper.get_current_platform
+def platform_matches?(platform)
+  Maze::Helper.get_current_platform == platform.downcase
 end
 
-Then(/^on (\w+) ([0-9.]+) and later, (.+)/) do |test_platform, test_version, step_text|
-  actual_platform = Maze::Helper.get_current_platform
-  unless %w[ios macos].include? actual_platform
```

Comment:
So was the problem here that it didn't account for watchOS?  Could we not have just allowed for that without refactoring it all, introducing a new problem?

Reply:
This code was checking the wrong thing, there is no need for the step to verify the return value from `Maze::Helper.get_current_platform`.

This step is now consistent with the other OS-conditional steps.

Reply:
Also I think using floats to represent OS versions is a bug - there are often more than 2 components to a version.

Reply:
Yeah true, although the whole mechanism is broken in a sense, when the OS version comes from a command line argument.  However, if what's here is working for you then that's all good.